### ISSUE TITLE
Feature request: support for IRCv3 client tag

### ISSUE THREAD

**Issue by Mikaela** [weechat/weechat#1785]
## Feature description

* https://ircv3.net/specs/client-tags/react.html

* [ ] Displaying reactions to messages
* [ ] Reacting to messages

I am not sure how it could work, in Matrix you can commonly and reply with `react whatever` to it.

I guess reactions could be shown just like normal emojis below the messages alongside a number of people reacting to it. I am not sure if they allow freeform text like Matrix, but for Matrix users it might be beneficial to allow that.

One method could be `/react <msgid> <reaction>` while that would be a bit clumsy.

**Comment by d3xt3r01** [comment]
I in wee-slack each message has an ID in which if somebody replies I see that ID and I can open a thread specific to that id and reply there. I guess that ID can be appended to each message only visible in weechat so I can `/react id message` ? 
Another usage is irccloud which supports this.

**Comment by mhoran** [comment]
There is now a [script](https://weechat.org/scripts/source/irc_react_inline.py.html/) that adds supports for the react tag to weechat. However, it would still be nice to have first class support for the react tag in weechat.

The script's `/react` command requires you to know the msgid of the message you'd like to react to. Compared to wee-slack, which allows reactions via "relative" line number (e.g. respond to the third most recent message via `3+:smile:`), it's a bit difficult to use.

While having an interface to react via text should probably be maintained, having some way to react via mouse might make this more ergonomic.

If weechat adds support for reactions, it would be good to have this exposed via the API as well.

**Comment by flashcode** [comment]
Yes agree, I started dev for this feature two years ago, but it's not finished.

The idea in my implementation is to be able to complete /react command with a line id, eg you could do such command to react with +1 on a message sent by "nick": `/react nick@07:56:02.187654 +1`.\
The `nick@07:56:02.187654` being a command parameter completed with the [Tab] key: you can complete multiple times with the latest messages displayed, by order from most recent to older one. Then with this the irc plugin would find the source msgid, the user doesn't have to know it or to write it.

And I'll check how this can be exposed in API and relay clients.
